mri/operators/fourier/utils.py

```diff
@@ -308,3 +308,37 @@ def check_if_fourier_op_uses_sense(fourier_op):
         return fourier_op.implementation.uses_sense
     else:
         return False
+
+
+def estimate_density_compensation(kspace_loc, volume_shape, num_iterations=10):
+    """ Utils function to obtain the density compensator for a
+    given set of kspace locations.
+
+    Parameters
+    ----------
+    kspace_loc: np.ndarray
+        the kspace locations
+    volume_shape: np.ndarray
+        the volume shape
+    num_iterations: int default 10
+        the number of iterations for density estimation
+    """
+    from .non_cartesian import NonCartesianFFT
+    from .non_cartesian import gpunufft_available
+    if gpunufft_available is False:
+        raise ValueError("gpuNUFFT is not available, cannot "
+                         "estimate the density compensation")
+    grid_op = NonCartesianFFT(
+        samples=kspace_loc,
+        shape=volume_shape,
+        implementation='gpuNUFFT',
+        osf=1,
+    )
+    density_comp = np.ones(kspace_loc.shape[0])
+    for _ in range(num_iterations):
+        density_comp = (
+                density_comp /
+                np.abs(grid_op.op(grid_op.adj_op(density_comp, True), True))
+        )
+    del grid_op
```

Comment:
why do you need this? shouldn't it be garbage collected by Python?

Reply:
Well, I had a memory build up, so I had to add this.
But it turned out to be a bug in gpuNUFFT, which I have fixed and will make a side release for, I will remove this

Reply:
ok can you add a `#TODO` explaining that this will need change?

Reply:
No No, this can be removed. This has no effect on memory, we just need to update gpuNUFFT which I already did. So this is kind of void at the moment, just was stale..

Reply:
wait I didn't get that, you are already using the new gpuNUFFT release?

Reply:
Well no, I will make a new release by end of day. But my point is that no matter whether we have the fix or not, this line does not help in anyway at all.. The underlying C++ and CUDA code had to free the memory, which wasnt, due to which we had a memory leak.